packages/odyssey-react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -50,28 +52,99 @@ export const Tooltip = withTheme(
   forwardRef<HTMLElement, TooltipProps>((props, ref) => {
     const { children, id, label, position = "top", ...rest } = props;
 
+    const tooltipRef = useRef<HTMLDivElement>(null);
     const oid = useOid(id);
     const omitProps = useOmit(rest);
     const tooltipClasses = useCx(styles.root, styles[`${position}Position`]);
     const clone = cloneElement(children, { "aria-describedby": oid });
+    const [coords, setCoords] = useState({});
+    const [timeoutFn, setTimeoutFn] = useState<NodeJS.Timeout | null>();
+    const { SpaceScale1 } = useTheme();
+    const remValue = Number(SpaceScale1.replace("rem", ""));
+    const pxValue =
+      remValue *
+      parseFloat(getComputedStyle(document.documentElement).fontSize);
+    const getPosition = (rect: DOMRect) => {
+      switch (position) {
+        default:
+        case "top":
+          return {
+            left: rect.x + rect.width / 2,
```

Comment:
I can't just replace `left` with `'inset-block-start'` it doesn't seem to work. 🤷

Reply:
switch looks great nice work! 🎉 

On a long term maintenance note, it would be good to simplify the body of the Tooltip component here.

We could extract as much of `getPosition`, `showTooltip` and `hideTooltip` into a dedicated `utils.ts` sibling module or a similar refactor effort. Let me know what you think might be possible we have a few mega components that are hundreds of lines long and it makes comprehension difficult need to strike the right balance

Reply:
I think everything 'surrounding' the `Box` should be another component so that the final structure in the file would look something like 
```
...
return (
<ToolTipContainer>
   <Box
        as="aside"
        {...omitProps}
        ref={ref}
        id={oid}
        className={tooltipClasses}
        role="tooltip"
        color={false}
        fontSize={false}
        fontWeight={false}
        lineHeight={false}
      >
        {label}
      </Box>
 </ToolTipContainer>
```

and the calculations for the 'coords' would done in hook, `useToolTipPosition` that would be called by the now `ToolTipContainer`